gridcapa-core-cc-post-processing-app/src/main/java/com/farao_community/farao/core_cc_post_processing/app/PostProcessingService.java

```diff
@@ -173,136 +169,140 @@ private void zipAndUploadLogs(List<byte[]> logList, String logFileName) {
-            File cgmFile = new File(cgmZipTmpDir + cgm.getFilename());
-            try {
-                FileUtils.copyInputStreamToFile(inputStream, cgmFile);
-            } catch (IOException e) {
-                throw new CoreCCPostProcessingInternalException("error while copying cgm to tmp folder", e);
-            }
-        });
-
-        // Zip tmp folder
-        byte[] cgmsZipResult = ZipUtil.zipDirectory(cgmZipTmpDir);
-        String targetCgmsFolderName = NamingRules.generateCgmZipName(localDate);
-        String targetCgmsFolderPath = NamingRules.generateOutputsDestinationPath(targetMinioFolder, targetCgmsFolderName);
+    private void uploadOfOutputToMinio(byte[] outputBytes, String outputPath) {
+        try (InputStream logZipIs = new ByteArrayInputStream(outputBytes)) {
+            minioAdapter.uploadOutput(outputPath, logZipIs);
+        } catch (IOException e) {
+            throw new CoreCCPostProcessingInternalException(String.format("Error occured while uploading file %s to MinIO", outputPath), e);
+        }
+    }
 
-        try (InputStream cgmZipIs = new ByteArrayInputStream(cgmsZipResult)) {
-            minioAdapter.uploadOutput(targetCgmsFolderPath, cgmZipIs);
+    private void zipCgmsAndSendToOutputs(String targetMinioFolder, Map<TaskDto, ProcessFileDto> cgms, LocalDate localDate, String correlationId, String timeInterval) {
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+            fillZipArchiveWithValidatedCgms(cgms, byteArrayOutputStream, localDate, correlationId, timeInterval);
+            byte[] cgmArchiveBytes = byteArrayOutputStream.toByteArray();
+            sendCgmsArchiveToOutputs(targetMinioFolder, localDate, cgmArchiveBytes);
         } catch (IOException e) {
             throw new CoreCCPostProcessingInternalException(String.format("Exception occurred while zipping CGMs of business day %s", localDate), e);
-        } finally {
-            ZipUtil.deletePath(Paths.get(cgmZipTmpDir)); //NOSONAR
         }
-
     }
 
-    private void zipCnesAndSendToOutputs(String targetMinioFolder, Map<TaskDto, ProcessFileDto> cnes, LocalDate localDate) {
-        String cneZipTmpDir = "/tmp/cnes_out/" + localDate.toString() + "/";
-
-        // Add all cnes from minio to tmp folder
-        cnes.values().stream()
-            .filter(processFileDto -> processFileDto.getProcessFileStatus().equals(ProcessFileStatus.VALIDATED))
-            .forEach(cne -> {
-                InputStream inputStream = minioAdapter.getFileFromFullPath(cne.getFilePath());
-                File cneFile = new File(cneZipTmpDir + cne.getFilename());
-                try {
-                    FileUtils.copyInputStreamToFile(inputStream, cneFile);
+    private void fillZipArchiveWithValidatedCgms(Map<TaskDto, ProcessFileDto> cgms, ByteArrayOutputStream outputStream, LocalDate localDate, String correlationId, String timeInterval) {
```

Comment:
The methods `fillZipArchiveWithValidatedCgms`, `fillZipArchiveWithValidatedCnes` and `fillZipArchiveWithValidatedRaoResults` look very similar. Maybe they could share some of their code?